client/landing/stepper/hooks/use-site-migration-status.ts

```diff
@@ -0,0 +1,151 @@
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { useEffect } from 'react';
+import wpcom from 'calypso/lib/wp';
+import type { SitePlugin } from 'calypso/data/plugins/types';
+
+interface Response {
+	plugins: SitePlugin[];
+}
+
+interface SkipStatus {
+	installation: boolean;
+	activation: boolean;
+}
+type SitePluginParam = Pick< SitePlugin, 'slug' | 'name' >;
+
+type StatusVariation = 'idle' | 'loading' | 'success' | 'error' | 'pending' | 'skipped';
+
+interface SiteMigrationStatus {
+	waitingSite: StatusVariation;
+	activatingPlugin: StatusVariation;
+	installingPlugin: StatusVariation;
+	completed: boolean;
+	error: Error | null;
+}
+
+const fetchPluginsForSite = async ( siteIdOrSlug: string ): Promise< Response > =>
+	wpcom.req.get( `/sites/${ siteIdOrSlug }/plugins?http_envelope=1`, {
+		apiNamespace: 'rest/v1.2',
+	} );
+
+const installPlugin = async ( siteIdOrSlug: string, pluginSlug: string ) =>
+	wpcom.req.post( {
+		path: `/sites/${ siteIdOrSlug }/plugins/${ pluginSlug }/install`,
+		apiNamespace: 'rest/v1.2',
+	} );
+
+const activatePlugin = async ( siteIdOrSlug: string, pluginName: string ) =>
+	wpcom.req.post( {
+		path: `/sites/${ siteIdOrSlug }/plugins/${ encodeURIComponent( pluginName ) }`,
+		apiNamespace: 'rest/v1.2',
+		body: {
+			active: true,
+		},
+	} );
+
+const usePluginStatus = ( pluginSlug: string, siteIdOrSlug?: string ) => {
+	return useQuery( {
+		queryKey: [ 'onboarding-site-plugin-status', siteIdOrSlug, pluginSlug ],
+		queryFn: () => fetchPluginsForSite( siteIdOrSlug! ),
+		enabled: !! siteIdOrSlug,
+		retry: true,
+		retryDelay: 2000,
+		select: ( data ) => {
+			return {
+				isActive: data.plugins.some( ( plugin ) => plugin.slug === pluginSlug && plugin.active ),
+				isInstalled: data.plugins.some( ( plugin ) => plugin.slug === pluginSlug ),
+			};
+		},
+	} );
+};
+
+const usePluginInstallation = ( pluginSlug: string, siteIdOrSlug: string ) => {
+	return useMutation( {
+		mutationKey: [ 'onboarding-site-plugin-installation', siteIdOrSlug, pluginSlug ],
+		mutationFn: async () => installPlugin( siteIdOrSlug, pluginSlug ),
+		retry: true,
+		retryDelay: 2000,
+	} );
+};
+
+const usePluginActivation = ( pluginName: string, siteIdOrSlug: string ) => {
+	return useMutation( {
+		mutationKey: [ 'onboarding-site-plugin-activation', siteIdOrSlug, pluginName ],
+		mutationFn: async () => activatePlugin( siteIdOrSlug, pluginName ),
+		retry: true,
+		retryDelay: 2000,
+	} );
+};
+
+export const useSiteMigrationStatus = (
```

Comment:
Is the plan to use this hook for all pre-migration actions (move to AT + plugins) or only for plugins?
If this will be used for all actions then the return key called `waitingSite` might be a little ambiguous, something like `waitingForPluginList` might be better.
If this is only for plugins, then instead of `useSiteMigrationStatus` this should be `useSiteMigrationPluginInstallationStatus` or something similar that signals the specific purpose.

Reply:
I renamed it to `usePluginAutoInstallation` `https://github.com/Automattic/wp-calypso/pull/89805/commits/5dba7177c82b2df52cbc784a0698dbcdc4c09bf0`, because there is another usePluginInstallation hook inside the same module that is exclusively installing a plugin.

and I also renamed `waitingSite` to `waitingPluginList` which is way better. 

I initially thought about the `waitingSite` because there is a "long" time waiting to have the endpoint return the site's plugin list, but the new name reflects better it. 

Thanks